src/bp/nlu-core/entities/microsoft-extractor/index.ts

```diff
@@ -0,0 +1,189 @@
+import { Recognizer, IModel, ModelResult } from '@microsoft/recognizers-text'
+
+import { NLU } from 'botpress/sdk'
+import _ from 'lodash'
+
+import { EntityExtractionResult, SystemEntityExtractor, KeyedItem } from '../../typings'
+import { SystemEntityCacheManager } from '../entity-cache-manager'
+import {
+  SupportedLangs,
+  SupportedLangsList,
+  GlobalRecognizers,
+  LanguageDependantRecognizers,
+  DucklingUnitMapping,
+  DucklingDateMappings,
+  DucklingTypeMappings,
+  langToCulture,
+  MicrosoftValues,
+  MicrosoftTimeValues,
+  MicrosoftValue
+} from './typings'
+interface MicrosoftParams {
+  lang: SupportedLangs
+  recognizers: any[]
+}
+
+const BATCH_SIZE = 10000
+
+// Further improvements:
+// 1- in _extractBatch, shift results ==> don't walk whole array n times (nlog(n) vs n2)
+
+export class MicrosoftEntityExtractor implements SystemEntityExtractor {
+  constructor(private _cache: SystemEntityCacheManager, private readonly logger?: NLU.Logger) {
+    this.logger = logger
+  }
+
+  public async configure() {
+    await this._cache.restoreCache()
+  }
+
+  public resetCache() {
+    this._cache.reset()
+  }
+
+  public async extractMultiple(
+    inputs: string[],
+    lang: string,
+    useCache?: boolean
+  ): Promise<EntityExtractionResult[][]> {
+    let options: MicrosoftParams
+
+    if (!SupportedLangsList.includes(lang)) {
```

Comment:
This is a if statement. Typically it should be tested. 

What you can do is use [jest](typeRecognizer) to mock the import of `Recognizers` and make sure the correct tokenizers were called.

Reply:
This is tested with the `Return nothing for unsupported lang` indeed I put numbers and dates in russian so it shouldn't match and thus I know the good group of recognizers was called